src/__tests__/browserSuites/telemetry.spec.js

```diff
@@ -20,147 +20,222 @@ const config = {
+      {
+        split_with_config: {
+          treatment: 'control',
+          config: null
+        },
+        always_on: {
+          treatment: 'control',
+          config: null
+        }
+      },
+      'Evaluations with config returned control states for invalid input.');
 
-    finish.next();
+    assert.equal(client.track('someTT', 'someEvent'), true, 'Event was queued');
+    assert.equal(client.track('someTT', null), false, 'Invalid input.');
 
-    return 200;
   });
 
-  assert.equal(client.getTreatment('always_on'), 'control', 'Non ready usage.');
-
-  await client.ready();
-
-  // treatments and results are only validated so we know for sure when the function was actually running to compare the metrics.
-  assert.equal(client.getTreatment('always_on'), 'on', 'Evaluation was correct.');
-  assert.equal(client.getTreatment('always_on', () => { }), 'control', 'We should return control with invalid input.');
-
-  assert.deepEqual(client.getTreatmentWithConfig('split_with_config'), {
-    treatment: 'o.n',
-    config: '{"color":"brown","dimensions":{"height":12,"width":14},"text":{"inner":"click me"}}'
-  }, 'Evaluation with config was correct.');
-  assert.deepEqual(client.getTreatmentWithConfig('split_with_config', () => { }), {
-    treatment: 'control',
-    config: null
-  }, 'Evaluation with config returned control state for invalid input.');
-
-  assert.deepEqual(client.getTreatments(['always_on', 'always_off']), { always_on: 'on', always_off: 'off' }, 'Evaluations were correct.');
-  assert.deepEqual(client.getTreatments(['always_on', 'always_off', null], () => { }), { always_on: 'control', always_off: 'control' }, 'We should return map of controls with invalid input.');
-
-  assert.deepEqual(client.getTreatmentsWithConfig(['split_with_config', 'always_on', null]),
-    {
-      split_with_config: {
-        treatment: 'o.n',
-        config: '{"color":"brown","dimensions":{"height":12,"width":14},"text":{"inner":"click me"}}'
-      },
-      always_on: {
-        treatment: 'on',
-        config: null
-      }
-    }
-    , 'Evaluations with config were correct.');
-  assert.deepEqual(client.getTreatmentsWithConfig(['split_with_config', 'always_on', null], () => { }),
-    {
-      split_with_config: {
-        treatment: 'control',
-        config: null
-      },
-      always_on: {
-        treatment: 'control',
-        config: null
-      }
+  // Flag sets
+  const baseConfig = {
+    core: {
+      authorizationKey: '<fake-token>',
+      key: 'nicolas@split.io'
     },
-    'Evaluations with config returned control states for invalid input.');
-
-  assert.equal(client.track('someTT', 'someEvent'), true, 'Event was queued');
-  assert.equal(client.track('someTT', null), false, 'Invalid input.');
+    urls: baseUrls,
+    scheduler: {
+      telemetryRefreshRate: 60
+    },
+    streamingEnabled: false
+  };
+
+  t.test(async (assert) => {
```

Comment:
You can replace lines 186 to 239 with the following, in order to remove duplicated code:
```
  t.test(async (assert) => {
    let factory;
    const splitFilters = [{ type: 'bySet', values: ['a', '_b', 'a', 'a', 'c', 'd', '_d'] }];

    fetchMock.get(baseUrls.sdk + '/mySegments/nicolas%40split.io', { status: 200, body: { 'mySegments': [] } });
    fetchMock.getOnce(baseUrls.sdk + '/splitChanges?since=-1&sets=a,c,d',  { status: 200, body: { splits: [], since: 1457552620999, till: 1457552620999 } });
    fetchMock.postOnce(baseUrls.telemetry + '/v1/metrics/config', (url, opts) => {
      const data = JSON.parse(opts.body);

      assert.true(data.tR > 0, 'timeUntilReady is larger than 0');
      assert.equal(data.fsT, 7, 'unique flag sets total should be 7');
      assert.equal(data.fsI, 4, 'flagset ignored should be 4');

      return 200;
    });

    fetchMock.postOnce(baseUrls.telemetry + '/v1/metrics/usage', (url, opts) => {
      const data = JSON.parse(opts.body);

      assert.deepEqual(data.mL.tf, [1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], 'Latencies stats');
      assert.deepEqual(data.mL.tfs, [1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], 'Latencies stats');
      assert.deepEqual(data.mL.tcf, [1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], 'Latencies stats');
      assert.deepEqual(data.mL.tcfs, [1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], 'Latencies stats');

      factory.client().destroy().then(() => {
        assert.end();
      });

      return 200;
    });
    fetchMock.postOnce(baseUrls.telemetry + '/v1/metrics/usage',  200);

    factory = SplitFactoryForTest({...baseConfig, sync: {splitFilters}});
    const client = factory.client();
    assert.deepEqual(client.getTreatmentsByFlagSet('a'),[]);
    assert.deepEqual(client.getTreatmentsByFlagSets(['a']),[]);
    assert.deepEqual(client.getTreatmentsWithConfigByFlagSet('a'),[]);
    assert.deepEqual(client.getTreatmentsWithConfigByFlagSets(['a']),[]);

  }, 'SDK with sets configured has sets information in config POST and evaluation by sets telemetry in stats POST');
```